Code/Framework/AzFramework/Platform/Linux/AzFramework/Input/Devices/Keyboard/InputDeviceKeyboard_xcb.cpp

```diff
@@ -0,0 +1,265 @@
+#include <xkbcommon/xkbcommon.h>
+#include <xkbcommon/xkbcommon-x11.h>
+
+namespace AzFramework
+{
+    class InputDeviceKeyboardXcb
+        : public InputDeviceKeyboard::Implementation
+        , public RawInputNotificationBusXcb::Handler
+    {
+    public:
+        AZ_CLASS_ALLOCATOR(InputDeviceKeyboardXcb, AZ::SystemAllocator, 0);
+
+        using InputDeviceKeyboard::Implementation::Implementation;
+        InputDeviceKeyboardXcb(InputDeviceKeyboard& inputDevice)
+            : InputDeviceKeyboard::Implementation(inputDevice)
+        {
+            RawInputNotificationBusXcb::Handler::BusConnect();
+
+            auto* connection = AzFramework::LinuxXcbConnectionManagerInterface::Get()->GetXcbConnection();
+
+            AZStd::unique_ptr<xcb_xkb_use_extension_reply_t, DeleterForFreeFn<::std::free>> xkbUseExtensionReply{
+                xcb_xkb_use_extension_reply(connection, xcb_xkb_use_extension(connection, 1, 0), nullptr)
+            };
+            m_coreDeviceId = xkb_x11_get_core_keyboard_device_id(connection);
+
+            m_xkbContext.reset(xkb_context_new(XKB_CONTEXT_NO_FLAGS));
+            m_xkbKeymap.reset(xkb_x11_keymap_new_from_device(m_xkbContext.get(), connection, m_coreDeviceId, XKB_KEYMAP_COMPILE_NO_FLAGS));
+            m_xkbState.reset(xkb_x11_state_new_from_device(m_xkbKeymap.get(), connection, m_coreDeviceId));
+        }
+
+        bool IsConnected() const override
+        {
+            auto* connection = AzFramework::LinuxXcbConnectionManagerInterface::Get()->GetXcbConnection();
+            return connection && !xcb_connection_has_error(connection);
+        }
+
+        bool HasTextEntryStarted() const override
+        {
+            return false;
+        }
+
+        void TextEntryStart(const InputDeviceKeyboard::VirtualKeyboardOptions& options) override
+        {
+        }
+
+        void TextEntryStop() override
+        {
+        }
+
+        void TickInputDevice() override
+        {
+            ProcessRawEventQueues();
+        }
+
+        void OnRawInputEvent(xcb_generic_event_t* event) override
+        {
+            switch (event->response_type & ~0x80)
+            {
+            case XCB_KEY_PRESS:
+                {
+                    auto* keyPress = reinterpret_cast<xcb_key_press_event_t*>(event);
+
+                    const InputChannelId* key = InputChannelFromKeyEvent(keyPress->detail);
+                    if (key)
+                    {
+                        QueueRawKeyEvent(*key, true);
+                    }
+                    break;
+                }
+            case XCB_KEY_RELEASE:
+                {
+                    auto* keyRelease = reinterpret_cast<xcb_key_release_event_t*>(event);
+
+                    const InputChannelId* key = InputChannelFromKeyEvent(keyRelease->detail);
+                    if (key)
+                    {
+                        QueueRawKeyEvent(*key, false);
+                    }
+                    break;
+                }
```

Comment:
The formatting with the curly braces isn't off here

Reply:
Fixed.